bigquery-connector-common/src/test/java/com/google/cloud/bigquery/connector/common/integration/CustomCredentialsIntegrationTest.java

```diff
@@ -60,4 +61,32 @@ public void testAccessTokenProvider() throws Exception {
     assertThat(table).isNotNull();
     assertThat(accessTokenProvider.getCallCount()).isEqualTo(2);
   }
+
+  @Test
+  public void testAccessTokenProvider_withConfig() {
+    BigQueryCredentialsSupplier credentialsSupplier =
+        new BigQueryCredentialsSupplier(
+            Optional.of(DefaultCredentialsDelegateAccessTokenProvider.class.getCanonicalName()),
+            Optional.of("some-configuration"),
+            Optional.empty(),
+            Optional.empty(),
+            Optional.empty(),
+            Optional.empty(),
+            Optional.empty(),
+            Optional.empty());
+    Credentials credentials = credentialsSupplier.getCredentials();
+    assertThat(credentials).isInstanceOf(AccessTokenProviderCredentials.class);
+    DefaultCredentialsDelegateAccessTokenProvider accessTokenProvider =
+        (DefaultCredentialsDelegateAccessTokenProvider)
+            ((AccessTokenProviderCredentials) credentials).getAccessTokenProvider();
+    assertThat(accessTokenProvider.getCallCount()).isEqualTo(0);
+    assertThat(accessTokenProvider.getConfig()).isEqualTo("some-configuration");
+
+    BigQueryOptions options = BigQueryOptions.newBuilder().setCredentials(credentials).build();
+    BigQuery bigQuery = options.getService();
+    // first call
+    Table table = bigQuery.getTable(TABLE_ID);
+    assertThat(table).isNotNull();
+    assertThat(accessTokenProvider.getCallCount()).isEqualTo(0);
```

Comment:
Expected value is 1, as `getAccessToken()` has been called once in order to make the call the BigQuery API

Reply:
Fixed that